src/util.cpp

```diff
@@ -1059,6 +1059,29 @@ QString NetworkUtil::FixAddress ( const QString& strAddress )
     return strAddress.simplified().replace ( " ", "" );
 }
 
+// Return whether the given HostAdress is within a private IP range
+// as per RFC 1918 & RFC 5735.
+bool NetworkUtil::IsPrivateNetworkIP ( const QHostAddress &qhAddr )
+{
+    // https://www.rfc-editor.org/rfc/rfc1918
+    // https://www.rfc-editor.org/rfc/rfc5735
+    QList<QPair<QHostAddress, int>> addresses =
+    {
+        QPair<QHostAddress, int>(QHostAddress("10.0.0.0"), 8),
+        QPair<QHostAddress, int>(QHostAddress("127.0.0.0"), 8),
+        QPair<QHostAddress, int>(QHostAddress("172.16.0.0"), 12),
+        QPair<QHostAddress, int>(QHostAddress("192.168.0.0"), 16),
+    };
+
+    int addressesSize = addresses.size();
+    for ( int iIdx = 0; iIdx < addressesSize; iIdx++ ) {
+        if (qhAddr.isInSubnet(addresses[iIdx])) {
+            return true;
+        }
+    }
```

Comment:
I have little to zero C++/Qt experience. There may well be better approaches to handle such list-based checks. I'm open to suggestions.

Reply:
You can simplify the loop using Qt's [foreach](https://doc.qt.io/qt-5/containers.html#foreach) - something like:

```
foreach ( auto item, addresses ) {
    if ( qhAddress.isInSubset( item ) ) {
        return true;
    }
}
```

(In new code I'd use standard C++'s [range-based for](https://en.cppreference.com/w/cpp/language/range-for), but Jamulus tends to use Qt's facilities.)

It'd also be good to declare `addresses` as static [constexpr](https://en.cppreference.com/w/cpp/language/constexpr), so it doesn't get constructed anew every time this bit of code runs.

Reply:
> You can simplify the loop using Qt's foreach - something like:

Thanks, done!

> It'd also be good to declare addresses as static constexpr, so it doesn't get constructed anew every time this bit of code runs.

Hrm, I understand the intention (and was even thinking about such a thing), but `constexpr` doesn't seem to work (`src/util.cpp:1068:54: error: the type 'const QList<QPair<QHostAddress, int> >' of 'constexpr' variable 'addresses' is not literal`). Per the docs it only works on literals (which I read as: int, char, etc.?) while I'm using a QList. I have now changed it to just `static`. Wouldn't this accomplish the same?